gen.py

```diff
@@ -165,38 +186,58 @@ def __init__(self, schema, lang):
+        """Returns the xsd definition for a given element from schemas defined in `self.tree` (or `self.tree2` if nothing found).
 
-        tag_name -- the name of the tag in the schema, e.g. 'complexType'
-        name_attribute -- the value of the 'name' attribute in the schema, ie.
-                          the name of the element/type etc. being described,
-                          e.g. iati-activities
+        Args:
+            tag_name (str): The name of the tag in the schema - will typically be 'element'.
+            name_attribute (str): The value of the 'name' attribute in the schema - i.e. the name of the element/type etc. being described, e.g. 'iati-activities'.
 
+        Returns:
+            None / lxml.etree._Element: The element tree representng the xsd definition for the given inputs. None if no match found.
         """
         schema_element = self.tree.find("xsd:{0}[@name='{1}']".format(tag_name, name_attribute), namespaces=namespaces)
         if schema_element is None:
             schema_element = self.tree2.find("xsd:{0}[@name='{1}']".format(tag_name, name_attribute), namespaces=namespaces)
         return schema_element
 
-    def schema_documentation(self, element, ref_element):
+
+    def schema_documentation(self, element, ref_element, type_element=None):
+        """Return a documention string for either a given ref_element (if not None) or an element.
+        Args:
+            element (lxml.etree._Element): An xsd element definition.
+            ref_element (lxml.etree._Element): An element that the `element` inherits properties definitions from (using the xsd `ref` inheritance).  If set to None, the documention string for the element is returned.
+            type_element (lxml.etree._Element): An element that the `element` inherits properties definitions from (using the xsd `type` inheritance). Defaults to None, implying element properties/definitions are defined within `element` or `ref_element`.
+
+        Returns:
+            str: The documentation string, extracted from the input ref_element or element.
+        """
+        if type_element is not None:
+            xsd_documentation = type_element.find(".//xsd:documentation", namespaces=namespaces)
+            if xsd_documentation is not None:
+                return type_element.find(".//xsd:documentation", namespaces=namespaces).text
+
         if ref_element is not None:
             xsd_docuementation = ref_element.find(".//xsd:documentation", namespaces=namespaces)
             if xsd_docuementation is not None:
                 return ref_element.find(".//xsd:documentation", namespaces=namespaces).text
         return element.find(".//xsd:documentation", namespaces=namespaces).text
 
 
-    def output_docs(self, element_name, path, element=None, minOccurs='', maxOccurs='', ref_element=None):
+    def output_docs(self, element_name, path, element=None, minOccurs='', maxOccurs='', ref_element=None, type_element=None):
         """
```

Comment:
Description could use bringing in mine with rest of linting style.

Reply:
Fixed in 963b481.